synnax/pkg/distribution/channel/create_test.go

```diff
@@ -143,4 +143,87 @@ var _ = Describe("Create", Ordered, func() {
 			Expect(ch.Key().Leaseholder()).To(Equal(aspen.Free))
 		})
 	})
+	Context("Updating a channel", func() {
+		var ch channel.Channel
+		var ch2 channel.Channel
+		BeforeEach(func() {
+			ch.Name = "SG0001"
+			ch.DataType = telem.Float64T
+			ch.Virtual = true
+			ch.Internal = false
+			ch.Leaseholder = core.Free
+
+			ch2.Rate = 5 * telem.Hz
+			ch2.Name = "SG0003"
+			ch2.DataType = telem.Float64T
+			ch2.Leaseholder = 1
+
+			Expect(services[1].Create(ctx, &ch)).To(Succeed())
+		})
+		It("Should update the channel name without error", func() {
+			ch.Name = "SG0002"
+			Expect(services[1].Create(ctx, &ch)).To(Succeed())
+			Expect(ch.Name).To(Equal("SG0002"))
+
+			var resChannels []channel.Channel
+			err := services[1].NewRetrieve().WhereKeys(ch.Key()).Entries(&resChannels).Exec(ctx, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resChannels).To(HaveLen(1))
+			Expect(resChannels[0].Name).To(Equal("SG0002"))
+		})
+		It("Should update the channel expression without error", func() {
+			ch.Expression = "sin(x)"
+			Expect(services[1].Create(ctx, &ch)).To(Succeed())
+			Expect(ch.Expression).To(Equal("sin(x)"))
+
+			var resChannels []channel.Channel
+			err := services[1].NewRetrieve().WhereKeys(ch.Key()).Entries(&resChannels).Exec(ctx, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resChannels).To(HaveLen(1))
+			Expect(resChannels[0].Expression).To(Equal("sin(x)"))
+		})
+		It("Should update the requires without error", func() {
+			ch.Requires = []channel.Key{ch2.Key()}
+			Expect(services[1].Create(ctx, &ch)).To(Succeed())
+			Expect(ch.Requires).To(ContainElement(ch2.Key()))
+
+			var resChannels []channel.Channel
+			err := services[1].NewRetrieve().WhereKeys(ch.Key()).Entries(&resChannels).Exec(ctx, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resChannels).To(HaveLen(1))
+			Expect(resChannels[0].Requires).To(ContainElement(ch2.Key()))
+		})
+		It("Should not update the channel if it already exists by name", func() {
+			ch.Name = "SG0003"
+			Expect(services[1].CreateIfNameDoesntExist(ctx, &ch)).To(Succeed())
+			Expect(ch.Name).To(Equal("SG0001"))
+
+			var resChannels []channel.Channel
+			err := services[1].NewRetrieve().WhereKeys(ch.Key()).Entries(&resChannels).Exec(ctx, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resChannels).To(HaveLen(1))
+			Expect(resChannels[0].Name).To(Equal("SG0001"))
+		})
+		It("Should not allow updating a non-virtual channel",
+			func() { // TODO: this test fails not related to the changes made to
```

Comment:
I don't understand this. Does it fail as in the channel does actually get updated? If so, we should fix this ASAP

Reply:
This channel doesn't actually get updated, it gets assigned a new key. This test should assert that the key of the old channel is not the same as the key of the new channel